app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
Dropzone currently provides a client-side drag & drop and file size check, which uses the application settings. You may want to look at that first. You may want to look at uploads_controller.rb to see how the server-side check is done for uploads: the upload is stored in a temporary place, verified, and then approved if it passes the file size check.